.circleci/config.yml

```diff
@@ -937,69 +903,129 @@ workflows:
           filters:
             branches:
-              only:
-                - auto
-      - docker-publish:
-          context: docker
+              only: auto
+      - lint-docker:
           filters:
             branches:
-              only:
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-      - code_coverage:
-          context: libra_ci
+              only: auto
+      - terraform:
           filters:
             branches:
-              only:
-                - /^test-[\d|.]+$/
-                - master
-      # Will build and push daily,
-      # or when relevant files change.
-      - docker-ci-build:
-          context: docker
-          base-image: circleci
+              only: auto
+      - prefetch-crates:
           filters:
             branches:
-              only:
-                - /^test-[\d|.]+$/
-                - master
+              only: auto
+      - lint:
+          requires:
+            - prefetch-crates
+          filters:
+            branches:
+              only: auto
+      - build-dev:
+          requires:
+            - prefetch-crates
+          filters:
+            branches:
+              only: auto
+      - run-e2e-test:
+          requires:
+            - prefetch-crates
+          filters:
+            branches:
+              only: auto
+      - run-unit-test:
+          requires:
+            - prefetch-crates
+          filters:
+            branches:
+              only: auto
+      - run-crypto-unit-test:
+          requires:
+            - prefetch-crates
+          filters:
+            branches:
+              only: auto
+      - build-benchmark:
+          requires:
+            - prefetch-crates
+          filters:
+            branches:
+              only: auto
+      - build-docs:
+          requires:
+            - lint
+          filters:
+            branches:
+              only: auto
       - docker-ci-build:
           base-image: circleci
           filters:
             branches:
-              ignore:
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-                - master
+              only: auto
       - docker-ci-build:
           base-image: centos
           filters:
             branches:
-              ignore:
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-                - master
+              only: auto
       - docker-ci-build:
           #sccache fails to execute in alpine.
           base-image: alpine
           filters:
             branches:
-              ignore:
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-                - master
+              only: auto
       - docker-ci-build:
           base-image: arch
           filters:
             branches:
-              ignore:
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-                - master
+              only: auto
+
+  pull-request-workflow:
```

Comment:
Do we need a separate "pull-request" and "bors" workflow? Might be easier to understand if these were the same, simply with the addition of the additional docker build/publish on the auto branch?

Reply:
Name collision prevention.   Separate checks will be easier to understand what roll they actually play.   I do wish I could group the jobs for reuse though.